### DISCUSSION TITLE
Can't connect to StrongSwan VPN with Android 14 native client

### DISCUSSION THREAD

**Discussion by glsmith86** [strongswan/strongswan#2102] [Q&A] [answered]
OS: Linux RT-AX56U 4.1.52 #1 SMP PREEMPT Sat Jan 20 12:44:31 EST 2024 armv7l ASUSWRT-Merlin
StrongSwan version: Linux strongSwan U5.9.11/K4.1.52

Main problem:
Feb 15 21:54:01 12[CFG] <2> looking for peer configs matching 192.168.1.1[%any]...vpn_external_ip[user]
Feb 15 21:54:01 14[CFG] <2> peer config "rw", ike match: 1052 (192.168.1.1...%any IKEv2)
Feb 15 21:54:01 14[CFG] <2>   local id match: 1 (ID_ANY: )
Feb 15 21:54:01 14[CFG] <2>   remote id match: 0 (ID_KEY_ID: 67:6c:73:6d:69:74:68)
Feb 15 21:54:01 12[CFG] <2> no matching peer config found                   

Configs & logs:

[strongswan.conf.txt](https://github.com/strongswan/strongswan/files/14302363/strongswan.conf.txt)
[swanctl.conf.txt](https://github.com/strongswan/strongswan/files/14302364/swanctl.conf.txt)
[charon.conf.txt](https://github.com/strongswan/strongswan/files/14302366/charon.conf.txt)
[ipsec.txt](https://github.com/strongswan/strongswan/files/14302380/ipsec.txt)
[strongswan_charonlog.txt](https://github.com/strongswan/strongswan/files/14302405/strongswan_charonlog.txt)

**Comment by tobiasbrunner** [comment, marked as answer]
While you apparently partially changed `glsmith` to `user` in some of the files you posted. Actually, right in the output you posted above we see that the client did not actually send `user` (as suggested by `vpn_external_ip[user]`) but `glsmith` as that's what the hex output `67:6c:73:6d:69:74:68` equals to.

On the same line we also see that the client sends the identity as type `ID_KEY_ID`. However, when you look at [the docs](https://docs.strongswan.org/docs/5.9/config/identityParsing.html) you'll see that strongSwan uses `ID_FQDN` for identities like `glsmith` or `user`, so these identities won't match. Either use an identity that both peers interpret the same way (e.g. an email address or an actual domain name), or explicitly configure the identity in strongSwan as key ID e.g. via `keyid:glsmith`.

**Reply by glsmith86** [reply]
@tobiasbrunner domain name version is working. I solved some other problems but I have an error what I can't solve by Google.
I can't ping from client the router but I can ping the client from router. I have read the Forwarding and Split-Tunneling page on strongswan.com. It doesn't help for me.

New configs:
[swanctl.conf.txt](https://github.com/strongswan/strongswan/files/14315898/swanctl.conf.txt)
[strongswan.conf.txt](https://github.com/strongswan/strongswan/files/14315899/strongswan.conf.txt)
[charon.conf.txt](https://github.com/strongswan/strongswan/files/14315900/charon.conf.txt)

ipsec statusall command
[ipsec_statusall.txt](https://github.com/strongswan/strongswan/files/14315905/ipsec_statusall.txt)

Client have 10.10.10.2 IP.
